### PR TITLE
*: support NotFillCache kv request option

### PR DESCRIPTION
**Pull request by tiancaiamao**
If TiDB send a request with NotFillCache option, the LRU cache of underlying
storage will not be touched. select statement with SQL_NO_CACHE will use it.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@coocood @disksing PTAL

File: executor/executor_test.go [hunk 1]
```diff
@@ -1951,12 +1953,19 @@ func (s *testSuite) TestIssue4024(c *C) {
...
 type checkRequestClient struct {
 	tikv.Client
-	priority pb.CommandPri
-	mu       struct {
+	priority     pb.CommandPri
+	notFillCache bool
+	mu           struct {
 		sync.RWMutex
-		turnOn bool
+		turnOn uint32
```
Comment (coocood) [inline executor/executor_test.go]: I think rename it to `checkFlags` and use bit operation is better.

File: executor/executor_test.go [hunk 2]
```diff
@@ -1951,58 +1953,79 @@ func (s *testSuite) TestIssue4024(c *C) {
...
 type checkRequestClient struct {
 	tikv.Client
-	priority pb.CommandPri
-	mu       struct {
+	priority     pb.CommandPri
+	notFillCache bool
+	mu           struct {
 		sync.RWMutex
-		turnOn bool
+		checkFlags uint32
```
Comment (disksing) [inline executor/executor_test.go]: Can't check both priority and notFillCache?

File: executor/executor_test.go [hunk 3]
```diff
@@ -1951,58 +1953,79 @@ func (s *testSuite) TestIssue4024(c *C) {
...
 }
 
 func (c *checkRequestClient) SendReq(ctx goctx.Context, addr string, req *tikvrpc.Request) (*tikvrpc.Response, error) {
 	resp, err := c.Client.SendReq(ctx, addr, req)
 	c.mu.RLock()
-	turnOn := c.mu.turnOn
+	checkFlags := c.mu.checkFlags
 	c.mu.RUnlock()
-	if turnOn {
+	if checkFlags == checkRequestPriority {
```
Comment (coocood) [inline executor/executor_test.go]: flag value should use bit operation like `checkFlags & checkRequestPriority > 0`

File: executor/executor_test.go [hunk 4]
```diff
@@ -1951,58 +1953,79 @@ func (s *testSuite) TestIssue4024(c *C) {
...
 		switch req.Type {
 		case tikvrpc.CmdCop:
 			if c.priority != req.Priority {
 				return nil, errors.New("fail to set priority")
 			}
 		}
+	} else if checkFlags == checkRequestNotFillCache {
```
Comment (coocood) [inline executor/executor_test.go]: ditto

File: executor/executor_test.go (hunk 2 above)
Comment (tiancaiamao) [reply executor/executor_test.go]: It's two separate test, in one testSuite.

File: executor/executor_test.go (hunk 3 above)
Comment (tiancaiamao) [reply executor/executor_test.go]: I don't need to check both of them now, so this is unnecessary.

**Comment by shenli** [comment]
@disksing @coocood PTAL

**Pull request merged**